courageous_comets/redis/messages.py

```diff
@@ -52,25 +52,41 @@ async def _get_messages_from_query(
 
 
 def build_search_scope(
-    id_: str,
+    guild_id: str,
+    ids: list[str] | None,
     scope: StatisticScope,
 ) -> FilterExpression:
     """
     Build a filter expression based on the given id and specified scope.
 
     Parameters
     ----------
-    id_ : str
-        The ID to search for.
+    guild_id: str
+        The ID of the guild to make the search.
+    ids : list[str] | None
+        The additional IDs to limit the search to.
     scope : courageous_comets.enums.StatisticScope
-        The scope to limit the search.
+        The scope of the additional IDs.
+
+    Notes
+    -----
+    If the scope of the search is courageous_comets.enums.StatisticScope.GUILD,
+        the `ids` are ignored.
 
     Returns
     -------
     redisvl.query.FilterExpression
         The redis filter expression for the specified scope.
     """
-    return Tag(scope) == id_
+    search_scope = Tag(StatisticScope.GUILD) == guild_id
+    # Ignore the other IDs as this would imply searching across multiple
+    # scopes.
+    if scope == StatisticScope.GUILD:
+        return search_scope
+
+    if ids:
+        search_scope = search_scope & (Tag(scope) == ids)
+    return search_scope
```

Comment:
```suggestion
    # Ignore the other IDs as this would imply searching across multiple scopes.
    if scope == StatisticScope.GUILD or not ids:
        return search_scope
    
    return search_scope & (Tag(scope) == ids)
```